src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/analysis/AnalysisDetailsPageIT.java

```diff
@@ -3,64 +3,427 @@
 
 import static junit.framework.TestCase.assertFalse;
-import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.*;
 import static org.junit.Assert.assertTrue;
 
+import ca.corefacility.bioinformatics.irida.ria.integration.utilities.FileUtilities;
+
+@ContextConfiguration(loader = AnnotationConfigContextLoader.class, classes = { IridaApiServicesConfig.class })
+@ActiveProfiles("it")
 @DatabaseSetup("/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisAdminView.xml")
-@Ignore
 public class AnalysisDetailsPageIT extends AbstractIridaUIITChromeDriver {
 	private static final Logger logger = LoggerFactory.getLogger(AnalysisDetailsPageIT.class);
+	private FileUtilities fileUtilities = new FileUtilities();
+
+	@Autowired
+	@Qualifier("outputFileBaseDirectory")
+	private Path outputFileBaseDirectory;
+
+	@Autowired
+	private IridaWorkflowLoaderService iridaWorkflowLoaderService;
+
+	@Before
+	// Tree file used by multiple tests
+	public void setFile() throws IOException {
+		fileUtilities.copyFileToDirectory(outputFileBaseDirectory, "src/test/resources/files/snp_tree.tree");
+	}
+
+	@Test
+	public void testAnalysisDetails() {
+		LoginPage.loginAsManager(driver());
+		AnalysisDetailsPage page = AnalysisDetailsPage.initPage(driver(), 4L, "settings/details");
+		assertTrue("Page title should equal", page.compareTabTitle("Details"));
+		assertEquals("There should be 7 labels for analysis details", 7, page.getNumberOfListItems());
+		// Analysis Description doesn't have a value
+		assertEquals("There should be only 6 values for these labels", 6, page.getNumberOfListItemValues());
+
+		String expectedAnalysisDetails[] = new String[] { "My Completed Submission", "4",
+				"SNVPhyl Phylogenomics Pipeline (1.0.1)", "MEDIUM", "Oct 6, 2013 10:01 AM", "a few seconds" };
+		assertTrue("The correct details are displayed for the analysis",
+				page.analysisDetailsEqual(expectedAnalysisDetails));
+	}
+
+	@Test
+	public void testBioHanselOutput() throws IOException {
+		fileUtilities.copyFileToDirectory(outputFileBaseDirectory, "src/test/resources/files/bio_hansel-results.json");
+
+		LoginPage.loginAsManager(driver());
+
+		AnalysisDetailsPage page = AnalysisDetailsPage.initPage(driver(), 12L, "");
+		assertTrue("Page title should equal", page.comparePageTitle("Bio Hansel Information"));
+
+		assertTrue("Has 5 list items for Bio Hansel Information", page.expectedNumberOfListItemsEqualsActual(5));
+
+		page = AnalysisDetailsPage.initPage(driver(), 12L, "output");
+		assertTrue("Page title should equal", page.comparePageTitle("Output File Preview"));
+		assertEquals("There should be one output file", 1, page.getNumberOfFilesDisplayed());
+		assertTrue("There should be exactly one download all files button", page.downloadAllFilesButtonVisible());
+		assertTrue("There should be a download button for the file that is displayed",
+				page.downloadOutputFileButtonVisible());
+	}
+
+	@Test
+	public void testDeleteAnalysis() {
```

Comment:
Is there way to test this better? It looks like we just see that there are 10 analyses (when there are more on another page), delete one, and then see that there are still 10. Could we be looking at page 2 instead and then assert 4 and 3?

Reply:
Updated in 6b29dcd95568eb2f6722f1be5f1d4fd201a9043f